pytensor/tensor/subtensor.py

```diff
@@ -1976,8 +1979,7 @@ def make_node(self, x, ilist):
             raise TypeError("index must be vector")
         if x_.type.ndim == 0:
             raise TypeError("cannot index into a scalar")
-        out_shape = (ilist_.type.shape[0],) + x_.type.shape[1:]
-        out_shape = tuple(1 if s == 1 else None for s in out_shape)
```

Comment:
Do you know why this information was being purposely discarded? It looks like it was introduced [here](https://github.com/jessegrabowski/pytensor/commit/4c02133ad4a844bcfd5830f0eccf2d7b27a2f560), but it's a big PR and I don't really get it.

Reply:
My guess is either something was broken elsewhere that got fixed since then, or it was just worries about changing too many things in a single sweep